### PR TITLE
Text in Voice: new VoiceChannel attributes

### PR DESCRIPTION
**Pull request by zhudotexe**
## Summary

This PR adds additional attributes that were previously found only on TextChannels to VoiceChannels (namely `nsfw`, `last_message_id`, and `slowmode_delay`). The API currently sends `last_message_id` and `rate_limit_per_user` for all voice channels regardless of whether or not the channel's guild has access to Text in Voice.

This PR also adds documentation about supplying `nsfw` and `slowmode_delay` to `VoiceChannel.edit()` - currently this is undefined behaviour but may be defined in the future (source: https://canary.discord.com/channels/808030843078836254/847513235536216085/914260051626164255). Since these attributes are exposed by the API I thought them relevant to add.

This PR also fixes an issue where typing events in VoiceChannels would not fire a `TYPING_BEGIN` event and adds a missing member cache population in `on_reaction_add`.

## Checklist

- [x] If code changes were made then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `black .`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -1009,6 +1040,31 @@ async def clone(
             {"bitrate": self.bitrate, "user_limit": self.user_limit}, name=name, reason=reason
         )
 
+    def is_nsfw(self) -> bool:
+        """:class:`bool`: Checks if the channel is NSFW."""
```
Comment (Victorsitou) [inline disnake/channel.py:1044]: Please use ``:return type: :class:`bool` `` as the current one would look like this:
image

note: I made a commit to change it in all docstrings [here](https://github.com/DisnakeDev/disnake/pull/153/commits/4457e3b90c6725df5f9c9d547e7acaa7b2798a91).

**Review by Victorsitou — APPROVED** [review]

File: disnake/channel.py [hunk 2]
```diff
@@ -1009,6 +1040,34 @@ async def clone(
...
+            or :meth:`fetch_message` with the :attr:`last_message_id`
+            attribute.
+
+        Returns
+        ---------
+        Optional[:class:`Message`]
+            The last message in this channel or ``None`` if not found.
+        """
+        return self._state._get_message(self.last_message_id) if self.last_message_id else None
+
```
Comment (shiftinv) [inline disnake/channel.py:1070]:
```suggestion

    def get_partial_message(self, message_id: int, /) -> PartialMessage:
        """Creates a :class:`PartialMessage` from the message ID.

        This is useful if you want to work with a message and only have its ID without
        doing an unnecessary API call.

        .. versionadded:: 2.3

        Parameters
        ------------
        message_id: :class:`int`
            The message ID to create a partial message for.

        Returns
        ---------
        :class:`PartialMessage`
            The partial message.
        """

        from .message import PartialMessage

        return PartialMessage(channel=self, id=message_id)
```

I haven't taken a close look, but mypy made me aware of the fact that `VoiceChannel.get_partial_message` didn't exist :P

Comment (zhudotexe) [reply disnake/channel.py:1070]: good catch!

**Review by EQUENOS — APPROVED** [review]

**Review by shiftinv — APPROVED** [review]
Unfortunately there are merge conflicts due to #176, but other than that, lgtm

**Pull request merged**